### ISSUE TITLE
Page margin problem

### ISSUE THREAD

**Issue by ustcfdm** [ElegantLaTeX/ElegantBook#49]
I have a few problems about margin.

1. The section title exceeds the content border, as indicated by the arrows in the figure below.
![image](https://user-images.githubusercontent.com/12667807/63614032-adf16700-c5a7-11e9-9ce2-c6d56d67e956.png)

2. In twoside mode, the left and right margins for odd and even page are inverse. As shown below, the right margin is wider for odd page, and the left margin is wider for even page. 
![image](https://user-images.githubusercontent.com/12667807/63614321-57d0f380-c5a8-11e9-944e-1420c6d213b2.png)
![image](https://user-images.githubusercontent.com/12667807/63614343-65867900-c5a8-11e9-9352-d065056e5c30.png)

3. If I manually set page margin, the header text doesn't align with header line, and the margin size is not what I specified. For example
`\geometry{top=1cm,bottom=1cm,inner=2.5cm,outer=1.5cm,includeall}`, it shows
![image](https://user-images.githubusercontent.com/12667807/63616176-9a94ca80-c5ac-11e9-8301-551573bf20e6.png)
![image](https://user-images.githubusercontent.com/12667807/63616212-a7b1b980-c5ac-11e9-92e8-af29c5b1f6f0.png)

**Comment by muzimuzhi** [comment]
> **1.** The section title exceeds the content border, as indicated by the arrows in the figure below.

This is a feature, as the template manually offset `2.45em` at the beginning of each `\section`-like commands.
https://github.com/ElegantLaTeX/ElegantBook/blob/ae0d420e4954f2cc35cba577384496ab86573758/elegantbook.cls#L366-L373

To get rid of this "hanging style" for section titles, you can remove the offset h-spaces in `cls` file, or copy and paste the section related source line and remove offset part.
```latex
\documentclass{elegantbook}
\titleformat{\section}[hang]{\bfseries}
  {\Large\bfseries{\color{structurecolor}\thesection}\enspace} % h-offset removed
  {1pt}
  {\color{structurecolor}\Large\bfseries\filright}
\begin{document}
...
\end{document}
```


> **2.** In twoside mode, the left and right margins for odd and even page are inverse. As shown below, the right margin is wider for odd page, and the left margin is wider for even page, which should be opposite.

This is the default setting of `geometry` package, which is 1:1 (left:right) for oneside mode and 2:3 for twoside. Check the doc of `geometry`, sec. 6.4. The template does not change left/right-margin settings.

To effectively and compatibly reset margins, see the following response to your third sub-question.


> **3.** If I manually set page margin, the header text doesn't align with header line, and the margin size is not what I specified.

This is the compatibility issue between `geometry` and `fancyhdr`, where 
 - `fancyhdr` stores a _constant_ head/foot-line width, but
 - any `geometry` setting after `fancyhdr` may change the width of text body, hence width of head/foot.

In short term, you can force `fancyhdr` to re-calculate ([ref](https://tex.stackexchange.com/a/402458/79060)), by
```latex
\geometry{top=1cm,bottom=1cm,inner=2.5cm,outer=1.5cm,includeall}
\fancyfootoffset{0pt}
```
In the long term, if users of the template often meet the similar problem, using another head/foot package like `titleeps` or `scrlayer-scrpage` which stores a variable (a macro which won't get its value until expanding) head/foot-line width is worth considering.

------

PS: Officially, I am not the maintainer of this repo, therefore my responses represent personal options only. And I know nothing about "why here is designed to be the current appearance/interface" except for reading and learning from source code, as well as documentations.

**Comment by ustcfdm** [comment]
@muzimuzhi The "hanging style" is a weird feature. I don't think it is a commonly used style, since I didn't see such a style before. I suggest removing this feature as a default setting. 
As for the page margin, what you said works well. However, I just upgrade to version 3.09, I got a new problem.  For example, I set
`\geometry{top=1cm,bottom=1cm,inner=2.5cm,outer=1.5cm,includeheadfoot}
\fancyfootoffset{0pt}`
The odd page number is now at inner side of the page, and outer side for even page number, which is supposed to be the opposite style. Is there a way to change it?
![image](https://user-images.githubusercontent.com/12667807/63632513-e66f6000-c5fc-11e9-866d-e38911c3043e.png)
![image](https://user-images.githubusercontent.com/12667807/63632515-f38c4f00-c5fc-11e9-84f8-f537ecbac971.png)

**Comment by muzimuzhi** [comment]
> The odd page number is now at inner side of the page, and outer side for even page number, which is supposed to be the opposite style.

This again, is supposed to be a "feature".
https://github.com/ElegantLaTeX/ElegantBook/blob/ae0d420e4954f2cc35cba577384496ab86573758/elegantbook.cls#L654-L664
 - `\ELEGANT@base` has value "base" or "hide", where "base" is its initial value.
 - In the linked source lines, page number (`\thepage`) is set to show at the right side of even pages (`ER`) and left side of odd pages (`OL`).

If you want to exchange these, switch `L` and `R` in the optional arguments of `\fancyhead` command.

**Comment by EthanDeng** [comment]
Actually, it's **not** a weird feature, it just make the section/subsection stand out. If you really don't like this weird feature, you can remove the `\hspace{-2.45em}` from the section settings part in `elegantlatex.cls` (line 360-370 in version 3.07).

**Comment by EthanDeng** [comment]
According to the default settings/behavior of `book` class, the left/right should switch, please refer the [new update](https://github.com/ElegantLaTeX/ElegantBook/commit/1b29cf4d0e48da41d8c31784be3d22742e3b0708).

Indeed, it's my fault, I switched left to right in version 3.09.

**Comment by ustcfdm** [comment]
> Actually, it's **not** a weird feature, it just make the section/subsection stand out. If you really don't like this weird feature, you can remove the `\hspace{-2.45em}` from the section settings part in `elegantlatex.cls` (line 360-370 in version 3.09).

In my opinion, there may be some trouble in this default hanging style. 

1. The section title is too close to the border. If I use twoside mode, when I print it out and bind it to a book, it may be not easy to see the section number. As shown below, the left margin is proper for me, but the section exceeds the border, which makes the margin narrow. 
![image](https://user-images.githubusercontent.com/12667807/63664723-89ef7a80-c78d-11e9-9d80-ac456b358b9a.png)
2. The section title will overlap with marginal notes (If there is). An example is shown below. 
![image](https://user-images.githubusercontent.com/12667807/63664803-100bc100-c78e-11e9-9858-838b904f04dc.png)
By the way, the marginal notes also exceed paper border. This is under default setting.
